src/index.ts

```diff
@@ -697,39 +688,49 @@ export async function startAnalysis<
-                    (async () => {
-                        await platform.waitForDevice(150, { abortSignal });
-
-                        if (emulator?.resetSnapshotName)
-                            await timeout(platform.resetDevice(emulator?.resetSnapshotName, { abortSignal }), {
-                                milliseconds: 5 * 60 * 1000,
-                                signal: abortSignal,
-                            });
-
-                        await platform.ensureDevice({ abortSignal });
+                await platform.waitForDevice(150, { abortSignal });
+
+                if (emulator?.resetSnapshotName)
+                    await timeout(platform.resetDevice(emulator?.resetSnapshotName, { abortSignal }), {
+                        milliseconds: 5 * 60 * 1000,
+                        signal: abortSignal,
+                    });
+
+                await platform.ensureDevice({ abortSignal });
+
+                if (
+                    !emulator?.resetSnapshotName &&
+                    targetOptions?.managed &&
+                    (analysisOptions?.capabilities as unknown as SupportedCapability<'android'> | undefined)?.includes(
+                        'frida'
+                    ) &&
+                    targetOptions?.managedEmulatorOptions.honeyData
+                ) {
+                    // We only really need to add this honey data once, if we have a snapshot to load.
+                    const honeyData = targetOptions?.managedEmulatorOptions.honeyData;
+                    if (honeyData.clipboard) await platform.setClipboard(honeyData.clipboard);
+                    if (honeyData.deviceName) await platform.setDeviceName(honeyData.deviceName);
+                }
 
-                        if (emulator?.managed && !emulator?.resetSnapshotName) {
-                            // The emulator is managed by us, so let’s take a snapshot after we ensured for the first time.
-                            const snapshotName = 'cyanoacrylate-ensured';
-                            await (
-                                platform as unknown as PlatformApi<'android', 'emulator', [], []>
-                            ).snapshotDeviceState(snapshotName, { abortSignal });
+                if (emulator?.managed && !emulator?.resetSnapshotName) {
+                    // The emulator is managed by us, so let’s take a snapshot after we ensured for the first time.
+                    const snapshotName = 'cyanoacrylate-ensured';
+                    await (platform as unknown as PlatformApi<'android', 'emulator', [], []>).snapshotDeviceState(
+                        snapshotName,
+                        { abortSignal }
+                    );
 
-                            // eslint-disable-next-line require-atomic-updates
-                            emulator.resetSnapshotName = snapshotName;
-                        }
-                    })(),
-                    abortSignal && rejectOnAbort(abortSignal),
```

Comment:
Can you explain why we don't need the `rejectOnAbort` anymore? Is this handled by the `abortSignal?.throwIfAborted();` and it was never needed?

Reply:
I realized that `rejectOnAbort()` contributed to the memory leakage, as we never de-registered the listener if there was no abort (and the promise never resolved). I mainly put it there, because we didn’t have so many methods accepting `abortSignals`, yet. But now that we abort down the line, that is better anyway (since rejecting here doesn’t stop any of the running processes).